src/libasr/codegen/wasm_to_x64.cpp

```diff
@@ -430,15 +414,12 @@ class X64Visitor : public WASMDecoder<X64Visitor>,
         emit_print_int_64(m_a, "print_i64");
         emit_print_double(m_a, "print_f64");
 
-        decoding_data_segment = true;
         // declare compile-time strings
+        std::string master_string = "";
         for (uint32_t i = 0; i < data_segments.size(); i++) {
-            offset = data_segments[i].insts_start_index;
-            decode_instructions();
-            std::string label = "string" + std::to_string(last_vis_i32_const);
-            label_to_str[label] = data_segments[i].text;
+            master_string += data_segments[i].text;
         }
-        decoding_data_segment = false;
+        label_to_str["master_string"] = master_string;
```

Comment:
**Note:** This approach works as we do not yet have support for arrays in the `wasm_x64` backend. When we start supporting compile-time arrays in assembly, we might need to change/update the `wasm` backend or `wasm_x64` backend.